File: q8/plc_test2_q8.py
def outer():
    x = 10
    print("outer: x = " + str(x))
    def f1():
        y = 20
        x = -10
        print("f1: x = " + str(x) + ", y = " + str(y))
        def f2():
            z = 30
            y = -20
            print("f2: x = " + str(x) + ", y = " + str(y) + ", z = " + str(z))
            def f3():
                w = 40
                z = -30
                print("f3: x = " + str(x) + ", y = " + str(y) + ", z = " + str(z) + ", w = " + str(w))
            f3()
        f2()
    f1()

outer()

File: q8/plc_test2_q8.js
function outer() {
    var x = 10;
    console.log("outer: x = " + x);
    function f1() {
        var y = 20;
        var x = -10;
        console.log("f1: x = " + x + ", y = " + y);
        function f2() {
            var z = 30;
            var y = -20;
            console.log("f2: x = " + x + ", y = " + y + ", z = " + z);
            function f3() {
                var w = 40;
                var z = -30;
                console.log("f3: x = " + x + ", y = " + y + ", z = " + z + ", w = " + w);
            }
            f3();
        }
        f2();
    }
    f1();
}

outer();